lasagne/layers/normalization.py

```diff
@@ -359,14 +359,32 @@ def batch_norm(layer, **kwargs):
     >>> [l.__class__.__name__ for l in get_all_layers(l2)]
     ['InputLayer', 'DenseLayer', 'BatchNormLayer', 'NonlinearityLayer']
     """
+    bn_name = kwargs.get('name', None)
+    wrapped_name = getattr(layer, 'name', None)
+    nonlin_name = None
+    # Create copy of kwargs passed onto BN layer without 'name'
+    bn_kwargs = dict((key, arg) for key, arg in kwargs.items()
+                     if key != 'name')
+    # If BN layer is explicitly named using kwargs,
+    # use it as a base name for the created Nonlinearity layer
+    if bn_name is not None:
+        nonlin_name = bn_name + '_nonlin'
+    # Else, use wrapped layer's name as base name for
+    # created BN and Nonlinearity layers
+    elif wrapped_name is not None:
+        bn_name = wrapped_name + '_bn'
+        nonlin_name = bn_name + '_nonlin'
+    # If neither BN layer is named, nor wrapped layer has name,
+    # leave all created layers unnamed
+
```

Comment:
Functionality is good, but I'd like to keep this shorter:
```python
bn_name = kwargs.pop('name', None) or (layer.name and layer.name + '_bn')
nonlin_name = bn_name and bn_name + '_nonlin'
```
Two lines instead of ten! I'd also move these two lines right before the respective layer is created. E.g., `nonlin_name` is only needed if we actually create a `NonlinearityLayer`.